java/src/main/java/com/google/openlocationcode/OpenLocationCode.java

```diff
@@ -507,22 +507,20 @@ public static boolean isValidCode(String code) {
     if (separatorPosition != code.lastIndexOf(SEPARATOR)) {
       return false;
     }
-
-    if (separatorPosition % 2 != 0) {
+    // There must be an even number of at most 8 characters before the separator.
+    if (separatorPosition % 2 != 0 || separatorPosition > SEPARATOR_POSITION) {
       return false;
     }
 
     // Check first two characters: only some values from the alphabet are permitted.
-    if (separatorPosition == 8) {
+    if (separatorPosition == SEPARATOR_POSITION) {
       // First latitude character can only have first 9 values.
-      Integer index0 = CODE_ALPHABET.indexOf(code.charAt(0));
-      if (index0 == null || index0 > 8) {
+      if (CODE_ALPHABET.indexOf(code.charAt(0)) > 8) {
```

Comment:
Minor code improvement to remove useless null checks since indexOf returns a primitive int (cannot be null) and invalid characters (when -1) are already validated after this.